### PR TITLE
[Exporter.Geneva][TLD] Fix serialization issue

### PR DESCRIPTION
**Pull request by utpilla**
## Changes
- Fix an issue which skipped `ToString()` when any non-primitive types are being serialized
- Fix another issue for serialization of types that implement `ISpanFormattable`

For significant contributions please make sure you have completed the following items:

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes

### CODE REVIEW

File: src/OpenTelemetry.Exporter.Geneva/TLDExporter/JsonSerializer.cs [hunk 1]
```diff
@@ -104,6 +104,22 @@ public static int SerializeString(byte[] buffer, int cursor, string value)
         return cursor;
     }
 
+#if NET6_0_OR_GREATER
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static int SerializeString(byte[] buffer, int cursor, ReadOnlySpan<char> value)
+    {
+        if (value == null)
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Exporter.Geneva/TLDExporter/JsonSerializer.cs:111]: `ReadOnlySpan` is a struct so you probably want to something like check `value.IsEmpty` here?

Comment (utpilla) [reply src/OpenTelemetry.Exporter.Geneva/TLDExporter/JsonSerializer.cs]: Fixed it.

File: src/OpenTelemetry.Exporter.Geneva/TLDExporter/JsonSerializer.cs [hunk 2]
```diff
@@ -316,18 +332,30 @@ public static int Serialize(byte[] buffer, int cursor, object obj)
                 return SerializeMap(buffer, cursor, v);
             case object[] v:
                 return SerializeArray(buffer, cursor, v);
+
 #if NET6_0_OR_GREATER
             case ISpanFormattable v:
                 tmp = stackalloc char[MAX_STACK_ALLOC_SIZE_IN_BYTES / sizeof(char)];
-                if (v.TryFormat(tmp, out charsWritten, default, CultureInfo.InvariantCulture))
+                if (v.TryFormat(tmp, out charsWritten, default, CultureInfo.InvariantCulture) && charsWritten > 0)
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Exporter.Geneva/TLDExporter/JsonSerializer.cs:339]: Seems a bit odd to have the `&& charsWritten > 0` here and the `value.IsEmpty` check above. Do we need both? Is there a particular test case you have in mind for this?

Comment (utpilla) [reply src/OpenTelemetry.Exporter.Geneva/TLDExporter/JsonSerializer.cs:339]: Right. The `value` check is not really needed as we already do a validation at the call site. I just left it in there for now as I thought I might reuse that function later on. I could remove that check and add a `Debug.Assert` there, if you'd like.

Comment (CodeBlanch) [reply src/OpenTelemetry.Exporter.Geneva/TLDExporter/JsonSerializer.cs:339]: I think switching to `Debug.Assert(!value.IsEmpty, "value was empty")` inside `SerializeString` would be fine. I'm not sure if we need the `&& charsWritten > 0` either TBH. If `TryFormat` returns `true` I think we trust that something was formatted. Up to you 😄

**Review by CodeBlanch — APPROVED** [review]
LGTM

**Pull request merged**